src/Transport/Sending/MessageSenderPool.cs

```diff
@@ -0,0 +1,95 @@
+﻿namespace NServiceBus.Transport.AzureServiceBus
+{
+    using System.Collections.Concurrent;
+    using System.Collections.Generic;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.ServiceBus;
+    using Microsoft.Azure.ServiceBus.Core;
+    using Microsoft.Azure.ServiceBus.Primitives;
+
+    class MessageSenderPool
+    {
+        public MessageSenderPool(ServiceBusConnectionStringBuilder connectionStringBuilder, ITokenProvider tokenProvider)
+        {
+            this.connectionStringBuilder = connectionStringBuilder;
+            this.tokenProvider = tokenProvider;
+
+            senders = new ConcurrentDictionary<(string, (ServiceBusConnection, string)), ConcurrentQueue<MessageSender>>();
+        }
+
+        public MessageSender GetMessageSender(string destination, (ServiceBusConnection connection, string path) receiverConnectionAndPath)
+        {
+            var sendersForDestination = senders.GetOrAdd((destination, receiverConnectionAndPath), _ => new ConcurrentQueue<MessageSender>());
+
+            if (!sendersForDestination.TryDequeue(out var sender) || sender.IsClosedOrClosing)
+            {
+                // Send-Via case
+                // TODO: replace with "if (receiverConnectionAndPath != (null, null))" when Resharper inspection 2018.2 is on TeamCity and is set to be default to support C# 7.3 syntax
+                // https://github.com/Particular/Deployment/issues/98
```

Comment:
This issue has been closed and therefore the TODO could be adressed

Reply:
Fixed in https://github.com/Particular/NServiceBus.Transport.AzureServiceBus/pull/29/commits/cc29335bff3bc7b143e8890120553fc62d018fa6